### PR TITLE
Fixing race condition issue with with options (warmUp) request

### PR DESCRIPTION
**Pull request by vishalnarkhede**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

**Review by ferhatelmas — APPROVED** [review]
LGTM

File: src/client.ts [hunk 1]
```diff
@@ -1116,11 +1116,19 @@ export class StreamChat<
       logger: this.logger,
     });
 
+    let warmUpPromise;
     if (this.options.warmUp) {
-      this.doAxiosRequest('options', this.baseURL + '/connect');
+      warmUpPromise = this.doAxiosRequest('options', this.baseURL + '/connect');
```
Comment (vicnicius) [inline src/client.ts:1121]: The start1, start2, wait2, wait1 logic looks a bit tricky to follow. Maybe something like the following would make it clearer:

```js
const warmUpHttp = () => this.options.warmUp ? this.doAxiosRequest('options', this.baseURL + '/connect') : Promise.resolve();
const [_, handshake] = await Promise.all([warmUpHttp(), this.wsConnection.connect()]);
```

Comment (vicnicius) [reply src/client.ts:1121]: Already merged. I was too slow x)

Comment (ferhatelmas) [reply src/client.ts:1121]: good points but `Promise.all` would make bundle bigger.

**Pull request merged**